irohad/ametsuchi/impl/postgres_block_index.cpp

```diff
@@ -5,125 +5,127 @@
+        commands.begin(),
+        commands.end(),
+        std::string{},
+        [&](auto query, const auto &cmd) {
+          auto transfer = getTransferAsset(cmd);
+          if (not transfer) {
+            return query;
+          }
+          const auto &src_id = transfer.value().srcAccountId();
+          const auto &dest_id = transfer.value().destAccountId();
 
-      return std::accumulate(
-          commands.begin(),
-          commands.end(),
-          true,
-          [&](auto &status, const auto &cmd) {
-            return visit_in_place(
-                cmd.get(),
-                [&](const shared_model::interface::TransferAsset &command) {
-                  status &=
-                      this->indexAccountIdHeight(command.srcAccountId(), height)
-                      & this->indexAccountIdHeight(command.destAccountId(),
-                                                   height);
+          query += makeAccountHeightIndex(src_id, height);
+          query += makeAccountHeightIndex(dest_id, height);
 
-                  auto ids = {account_id,
-                              command.srcAccountId(),
-                              command.destAccountId()};
-                  auto &asset_id = command.assetId();
-                  // flat map accounts to unindexed keys
-                  boost::for_each(ids, [&](const auto &id) {
-                    soci::statement st =
-                        (sql_.prepare
-                             << "INSERT INTO index_by_id_height_asset(id, "
-                                "height, asset_id, "
-                                "index) "
-                                "VALUES (:id, :height, :asset_id, :index)",
-                         soci::use(id),
-                         soci::use(height),
-                         soci::use(asset_id),
-                         soci::use(index));
-                    status &= execute(st);
-                  });
-                  return status;
-                },
-                [&](const auto &command) { return true; });
-          });
-    }
+          const auto ids = {account_id, src_id, dest_id};
+          const auto &asset_id = transfer.value().assetId();
+          // flat map accounts to unindexed keys
+          for (const auto &id : ids) {
+            boost::format base(
+                "INSERT INTO index_by_id_height_asset(id, "
+                "height, asset_id, "
+                "index) "
+                "VALUES ('%s', '%s', '%s', '%s');");
+            query += (base % id % height % asset_id % index).str();
+          }
+          return query;
+        });
+  }
+}  // namespace
+
+namespace iroha {
+  namespace ametsuchi {
+    PostgresBlockIndex::PostgresBlockIndex(soci::session &sql)
+        : sql_(sql), log_(logger::log("PostgresBlockIndex")) {}
 
     void PostgresBlockIndex::index(
         const shared_model::interface::Block &block) {
       const auto &height = std::to_string(block.height());
```

Comment:
Why stringifying the height, if it can be passed directly as `HeightType` and then inserted into boost format as `uint` value?